cases/models.py

```diff
@@ -174,47 +174,60 @@ def get_merged_into_cases(self, cases):
         if not case_ids:
             return {}
 
-        # Even though merging actions should not have an edited 'time', we need to
-        # use 'time' rather than 'created' for the merged 'at'.
-        # This is because in other action queries we use 'time' and we should be consistent,
-        # especially as there is a small delta between the values of 'time' and 'created' for
-        # a newly created action.
-
         query = self.raw(
             """WITH RECURSIVE cte AS (
                  SELECT
                    c.id,
-                   c.merged_into_id
+                   c.merged_into_id,
+                   last_merge.time
                  FROM
                    cases_case c
+                   JOIN (
+                     SELECT m.mergee_id, m.merged_into_id, MAX(m.time) AS time
+                     FROM cases_mergerecord m
+                     WHERE m.unmerge IS NOT TRUE
+                     GROUP BY m.mergee_id, m.merged_into_id
+                   ) AS last_merge ON last_merge.merged_into_id = c.merged_into_id
+                   AND last_merge.mergee_id = c.id
```

Comment:
Ah, so is this like this because we can't go just on the MergeRecords because there might be an unmerge record later than the latest merge record. Hmm. I don't know if this is any better or not, but whilst thinking about this, I came up with the following that only looks at the MergeRecord table, picking the 'latest' MergeRecord for each:
```
            """WITH RECURSIVE cte AS (
                 SELECT
                   m.mergee_id AS id,
                   m.merged_into_id,
                   m.time
                 FROM
                   cases_mergerecord m
                 WHERE
                   NOT m.unmerge
                   AND (SELECT id FROM cases_mergerecord m2 where m2.id>m.id AND m2.mergee_id=m.mergee_id AND m2.merged_into_id=m.merged_into_id) IS NULL -- no later MergeRecord
                   AND m.mergee_id IN (%s)
                 UNION
                 SELECT
                   m.mergee_id AS id,
                   m.merged_into_id,
                   m.time
                 FROM
                   cases_mergerecord m
                   JOIN cte ON m.mergee_id = cte.merged_into_id
                 WHERE
                   NOT m.unmerge
                   AND (SELECT id FROM cases_mergerecord m2 where m2.id>m.id AND m2.mergee_id=m.mergee_id AND m2.merged_into_id=m.merged_into_id) IS NULL -- no later MergeRecord
               )
               SELECT * FROM cte
            """
```
Tests still pass with this, but not sure if that's enough. Perhaps you could try it at your side with some data and see if you think it's any better.

Reply:
No idea about performance but does seem a bit easier to read; made the swap (with some small edits).